src/mp-instance.js

```diff
@@ -590,7 +590,7 @@ export default function mParticleInstance(instanceName) {
                     Constants.NativeSdkPaths.Upload
                 );
             } else {
-                self._APIClient.uploader.prepareAndUpload(false, false);
+                self._APIClient.uploader?.prepareAndUpload(false, false);
```

Comment:
Can you write a test that catches this?

Reply:
just added the test case, thnx!

Reply:
there exists a case where _APIClient won't yet be available, when there is a /config call in a self hosted world, so let's add a ? to _APIClient also: 
```suggestion
                self._APIClient?.uploader?.prepareAndUpload(false, false);
```